lib/Parse/ParseDecl.cpp

```diff
@@ -1611,6 +1611,14 @@ bool Parser::parseNewDeclAttribute(DeclAttributes &Attributes, SourceLoc AtLoc,
     DiscardAttribute = true;
   }
 
+  // If this attribute is only permitted when distributed is enabled, reject it.
+  if (DeclAttribute::isDistributedOnly(DK) &&
+      !shouldParseExperimentalDistributed()) {
+    diagnose(Loc, diag::attr_requires_distributed, AttrName,
+             DeclAttribute::isDeclModifier(DK));
+    DiscardAttribute = true;
+  }
+
```

Comment:
I feel like this is the only thing that's actually needed to solve the issue. Everything else is just for availability things.

Reply:
I'll remove the availability support from this PR, thanks